mypy/partially_defined.py

```diff
@@ -380,7 +380,7 @@ def visit_name_expr(self, o: NameExpr) -> None:
             self.tracker.record_definition(o.name)
         elif self.tracker.is_defined_in_different_branch(o.name):
             # A variable is defined in one branch but used in a different branch.
-            self.msg.var_used_before_def(o.name, o)
+            self.msg.variable_may_be_undefined(o.name, o)
```

Comment:
Can you assign the error code depending on whether we are currently inside any loop? I think it would be better, e.g. use `self.loop_depth` like we do in semanal.py

Reply:
Yep, that's better!